### ISSUE TITLE
[ODB] Resume upload fails to upload onenote file

### ISSUE THREAD

**Issue by joewutaiwan** [OneDrive/onedrive-api-docs#1078]
#### Category
- [ ] Question
- [ ] Documentation issue
- [x] Bug

Hi all,

Problem 1:
step 1: create a onenote in drive with pages like HELLO
so the onenote folder would contain HELLO.one file

step 2: use resume upload API to upload HELLO.one (create session then upload)
then keeps returning
```
"error":{
"code":"serviceNotAvailable",
"message":"The service is not available. Try the request again after a delay. 
There may be a Retry-After header."}
```

Problem 2
step 1: create a onenote in drive 

step 2: upload by simple upload TEST.one
> Works fine, I can open TEST page in onenote folder

step 3: upload by resume upload (create session then upload) TEST.one

> The TEST page is corrupted

It looks like something broken when using resume upload

Thank you.

**Comment by chackman** [comment]
Thanks for your feedback.  From the failing request can you please provide the following response headers corresponding to your request(s):
> - Date (in UTC, please)
> - request-id
> - SPRequestGuid (for requests made to OneDrive for Business)
>
> _(DELETE THIS PARAGRAPH AFTER READING)_
>

**Comment by joewutaiwan** [comment]
**Case 1 503 serviceNotAvailable problem**

a.

- Date: Fri, 14 Jun 2019 07:29:35 GMT
- request-id: 1271e69e-d01b-9000-fed2-cedd6e49e798
- SPRequestGuid: 1271e69e-d01b-9000-fed2-cedd6e49e798


b.

- add a onenote NoteBook 
- cression upload session for an default onenote file under onenote NoteBook (for example: Pages.one)

c.

- POST /drives/{driveId}/items/{itemId}/createUploadSession

d.

- Successfully create this session for upload


**case 2 corrupted one file**

a. I directly offer the corrupted file id 
- name: 789.one
- id: 012DOYZZUSJKKETPM65VC3GY6JRCUHWAIG
- ctime: 2019-06-14T07:26:43Z
- mtime_str: 2019-06-14T07:26:45Z

b.

- add a onenote NoteBook 
- use Upload large files with an upload session to upload a downloaded onenote page file
- can't open in onenote online with the uploaded onenote file, but works fine with simple upload API

c.

- POST /drives/{driveId}/items/{itemId}/createUploadSession
- PUT upload_url

d.

- Successfully create this onenote file

**Comment by chackman** [comment]
1. With the 503, that is an expected return given the state of the service at that point in time; wait for the time specified in the Retry-After response header, and then retry.
2. Can you please provide a request ID corresponding to the upload of that particular file?

**Comment by joewutaiwan** [comment]
Hi @chackman 

1. With this 503 error, there is no Retry-After header in response, should I bring some specific header in request?

2. Here you are:

```
name: 999.one, id: 012DOYZZSWREHW5V35HVGI62TO2LUY6CRY,
SPRequestGuid: 59b1e99e-5027-8000-fed2-c3a77a53f242
request-id: 59b1e99e-5027-8000-fed2-c3a77a53f242
Date: Mon, 24 Jun 2019 09:54:41 GMT
```

Thank you for further assistance.

**Comment by chackman** [comment]
I can't find the logs, so could you please help clarify something -

If the request is through Microsoft Graph, could you please clarify which [service root endpoint](https://docs.microsoft.com/en-us/graph/deployments#microsoft-graph-and-graph-explorer-service-root-endpoints) the call was through?  For example, "graph.microsoft.com", "graph.microsoft.de", etc.

If the request is directly through SharePoint, could you please clarify the domain name of the endpoint (without tenant name)?  For example, "sharepoint.com", "sharepoint.de", etc.

Regarding the lack of a Retry-After header, you shouldn't need to provide additional information in the request.  It might be a bug.

**Comment by joewutaiwan** [comment]
Hi @chackman 

I test this on the following domain:

`{tenant name}-my.sharepoint.com`

Thank you for help!

**Comment by chackman** [comment]
I've tried looking, no luck.  Could you please provide correlation ID's and times for successful requests during the same time period for the same tenant?

**Comment by joewutaiwan** [comment]
Hi @chackman 

Thanks for reply,

1. I don't keep any correlation ID's and times for successful requests during that time.
But this problem could be reproduced any time, 
so how could I offer header information to let you find the specific error request?
In previous experience, the **request-id** and **Date** is enough.

2. Is possible to reproduce this problem in your environment?
```
step 1: create a onenote and page TEST.one on OneDrive in O365
step 3: download TEST.one
step 3: upload by simple upload request TEST.one  -> works fine
step 4: upload by resume upload (create session then upload) TEST.one

The TEST page is broken after step 4
```

Thanks alot.

**Comment by chackman** [comment]
Sorry, I was unable to reproduce this.

**Comment by chackman** [comment]
Could you provide the full tenant name?  I want to try again to see if I can get logs.

**Comment by joewutaiwan** [comment]
Hi @chackman ,

According to my latest testing, the second issue has been fixed, both simple upload and resume upload work properly for onenote file.

However the first issue is still in problem: failed to create resume upload session, after I opened that onenote file. And here is the response header, it keeps to tell me retry but there is no Retry-after header.

`POST /drives/{driveId}/items/{itemId}/createUploadSession`
```
503
{"error":{"code":"serviceNotAvailable",
"message":"The service is not available. Try the request again after a delay. 
There may be a Retry-After header."}}

request-id: 8cecf29e-200f-9000-ddad-2f29013c2965
Date: Tue, 23 Jul 2019 02:14:35 GMT
tenant name: synologydst
```

thank you for help

**Comment by patrick-rodgers** [comment]
As part of a repository clean up effort we are closing older issues. If this issue remains, please: open a new issue, reference this issue, and provide any additional details that may help in resolution. Thank you for your understanding as we work to improve our responsiveness.
